File: abc083/abc083_b.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
  int n,a,b;
  cin >> n >> a >> b;

  int ans=0;
  for (int i = 1; i <= n; i++)
  {
    int num=i,sum=0;
    while(num!=0){
      sum+=num%10;
      num/=10;
    }
    if(sum>=a && sum<=b) ans+=i;
  }
  cout << ans << endl;
}

File: abc083/abc083_b.rb
n,a,b = gets.split.map(&:to_i)
ans=0
1.upto(n) do |i|
  num=i
  sum=0
  while num!=0
    sum+=num%10
    num/=10
  end
  if sum>=a && sum<=b
    ans+=i
  end
end
puts ans